test/widgets/emoji_reaction_test.dart

```diff
@@ -39,30 +42,37 @@ void main() {
 
     Future<void> setupChipsInBox(WidgetTester tester, {
       required List<Reaction> reactions,
-      double? width,
-      TextDirection? textDirection,
+      double width = 245.0, // (seen in context on an iPhone 13 Pro)
+      TextDirection textDirection = TextDirection.ltr,
     }) async {
       final message = eg.streamMessage(reactions: reactions);
 
-      await tester.pumpWidget(
-        MaterialApp(
-          home: Directionality(
-            textDirection: textDirection ?? TextDirection.ltr,
-            child: GlobalStoreWidget(
-              child: PerAccountStoreWidget(
-                accountId: eg.selfAccount.id,
-                child: Center(
-                  child: ColoredBox(
-                    color: Colors.white,
-                    child: SizedBox(
-                      width: width ?? 245.0, // (seen in context on an iPhone 13 Pro)
-                      child: ReactionChipsList(
-                        messageId: message.id,
-                        reactions: message.reactions!,
-                      )))))))));
-
-      // global store, per-account store
-      await tester.pumpAndSettle();
+      final locale = switch (textDirection) {
+        TextDirection.ltr => const Locale('en'),
+        TextDirection.rtl => const Locale('ar'),
+      };
+      tester.platformDispatcher.localeTestValue = locale;
+      tester.platformDispatcher.localesTestValue = [locale];
+      addTearDown(tester.platformDispatcher.clearLocaleTestValue);
+      addTearDown(tester.platformDispatcher.clearLocalesTestValue);
+
+      await tester.pumpWidget(const ZulipApp());
```

Comment:
This commit turns out to make these tests quite a bit slower:
```
  # before:
$ time flutter test test/widgets/emoji_reaction_test.dart
00:05 +160: All tests passed!                                                            

time: 6.078s wall (7.313s u, 0.940s s)


  # after:
$ time flutter test test/widgets/emoji_reaction_test.dart
00:09 +160: All tests passed!                                                            

time: 10.528s wall (12.424s u, 1.321s s)
```

The slowdown happens with this commit:
fa3f012e9 emoji_reaction test: Set up whole ZulipApp, not just MaterialApp

It'd be good to try to pin down where that time is going. The strategy of reusing ZulipApp in tests makes a lot of sense in principle, so we're likely to try using it in more and more tests… and I'd like to figure out how to do so without slowing all our tests down by 75%.

The strategy I'd try is to just start commenting out parts of ZulipApp's functionality, and rerunning this one test file, and see which pieces are contributing the time.

Then once we know which pieces are at issue, we can see about finding hacks for tests to cut those out, or strip them down to a minimum.